### PR TITLE
Onboarding tests filtered by language, environment(dev,pro) and weblog

### PR DESCRIPTION
**Pull request by robertomonteromiguel**
Push data results to system-tests-dashboard

## Reviewer checklist

* [ ] If this PR modifies anything else than strictly the default scenario, then remove the `run-default-scenario` label
* [ ] CI is green
   * [ ] If not, failing jobs are not related to this change (and you are 100% sure about this statement)

## Workflow

1. ⚠️⚠️ Create your PR as draft
2. Follow the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
3. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
4. Mark it as ready for review

> **_NOTE:_**  By default in PR only default scenario tests will be launched. Remove the `run-default-scenario` label to run all scenarios ([more info](https://datadoghq.atlassian.net/wiki/spaces/APMINT/pages/2866381467/CI+Workflow+Github+Actions))

Once your PR is reviewed, you can merge it! :heart:

### CODE REVIEW

File: utils/_context/core.py [hunk 1]
```diff
@@ -70,8 +78,12 @@ def serialize(self):
             "appsec_rules_file": self.appsec_rules_file or "*default*",
             "uds_socket": self.uds_socket,
             "scenario": self.scenario,
+            "components": self.components,
```
Comment (cbeauchesne) [inline utils/_context/core.py:81]: Either using components for all components, or flatten everything.

File: utils/_context/virtual_machines.py [hunk 2]
```diff
@@ -188,6 +175,19 @@ def start(self):
     def set_ip(self, instance_ip):
         self.ip = instance_ip
 
+    def set_components_json(self, components_json):
```
Comment (cbeauchesne) [inline utils/_context/virtual_machines.py:178]: Moving `get_component` logic inside `set_components_json`, rename it `set_components(self, json_source)`, and get result using `TestedVirtualMachine.components`.

File: utils/_context/_scenarios.py [hunk 3]
```diff
@@ -687,17 +695,73 @@ def __init__(self, name, doc) -> None:
         self.stack = None
         self.provision_vms = []
         self.provision_vm_names = []
+        self.onboarding_components = {}
+        self.onboarding_tests_metadata = {}
 
     def configure(self, replay):
```
Comment (cbeauchesne) [inline utils/_context/_scenarios.py:701]:
```suggestion
    def configure(self, option):
            super().configure(option)
            self._library = LibraryVersion(option.obd_library, "0.0")
            self._env = option.obd_env
            self._weblog = option.obd_weblog
```

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -108,51 +108,77 @@ onboarding_nodejs:
...
       matrix:
-        - TEST_LIBRARY: [nodejs]
+        - ONBOARDING_FILTER_ENV: [dev,prod]
           SCENARIO: [ONBOARDING_HOST,ONBOARDING_HOST_CONTAINER,ONBOARDING_CONTAINER, ONBOARDING_HOST_AUTO_INSTALL, ONBOARDING_HOST_CONTAINER_AUTO_INSTALL, ONBOARDING_CONTAINER_AUTO_INSTALL] 
+          ONBOARDING_FILTER_WEBLOG: [test-app-nodejs]
   script:
       - echo "SCENARIO=${SCENARIO}" >> .env_ci && echo "TEST_LIBRARY=${TEST_LIBRARY}" >> .env_ci
+      - echo "ENV=${ONBOARDING_FILTER_ENV}" >> .env_ci && echo "WEBLOG=${ONBOARDING_FILTER_WEBLOG}" >> .env_ci
       - ./build.sh -i runner
       - ./run.sh $SCENARIO
```
Comment (cbeauchesne) [inline .gitlab-ci.yml:122]:
```suggestion
      - ./run.sh $SCENARIO --obd-weblog=${ONBOARDING_FILTER_WEBLOG} --obd-env=${ONBOARDING_FILTER_ENV}
```

File: utils/_context/_scenarios.py [hunk 5]
```diff
@@ -687,17 +695,73 @@ def __init__(self, name, doc) -> None:
...
     def library(self):
         return LibraryVersion(os.getenv("TEST_LIBRARY"), "0.0")
 
+    @property
+    def components(self):
+        return self.onboarding_components
+
+    @property
+    def parametrized_tests_metadata(self):
+        myTest = {"hola": "adios", "hello": "goodby"}
```
Comment (cbeauchesne) [inline utils/_context/_scenarios.py:719]: Hello world :)

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
in `pytest_addoption`, add options `--obd-weblog`, `--obd-library`, `--obd-env`.

File: utils/_context/core.py [hunk 6]
```diff
@@ -70,8 +78,12 @@ def serialize(self):
...
         }
 
+        if self.parametrized_tests_metadata:
```
Comment (cbeauchesne) [inline utils/_context/core.py:84]: suggestion : 

```suggestion
    results |= self.components
```

And make any needed change on this object later.

**Review by cbeauchesne — COMMENTED** [review]
in `pytest_addoption`, add options `--obd-weblog`, `--obd-library`, `--obd-env`.

File: utils/_context/core.py (hunk 6 above)
Comment (robertomonteromiguel) [reply utils/_context/core.py:84]: applied for components, no for parametrized_tests_metadata

**Comment by cbeauchesne** [comment]
Do you need a review ?

**Pull request merged**